packages/common/doHSClient/src/doHSClient.ts

```
@@ -0,0 +1,76 @@
+}) => {
+  logger.info("Fetching DoHS API data for ODS codes", {odsCodes})
+
+  // Collect all valid ODS codes into an array
+  const validOdsCodes = [
+    odsCodes.prescribingOrganization,
+    odsCodes.nominatedPerformer,
+    ...(odsCodes.dispensingOrganizations ?? []) // Spread array for multiple dispensing orgs
+  ].filter(Boolean) as Array<string>
+
+  if (validOdsCodes.length === 0) {
+    throw new Error("At least one ODS Code is required for DoHS API request")
+  }
+
+  // Use APIGEE_PTL_DOHS_API_KEY if available, otherwise fall back to apigeeApiKey
+  const effectiveApiKey = apigeePtlDoHSApiKey ?? apigeeApiKey
+  if (!effectiveApiKey) {
+    throw new Error("Apigee API Key environment variable is not set")
+  }
+  if (!apigeeDoHSEndpoint) {
+    throw new Error("DoHS API endpoint environment variable is not set")
+  }
+
+  // Construct filter query for multiple ODS codes
+  const odsFilter = validOdsCodes.map((code) => `ODSCode eq '${code}'`).join(" or ")
+
+  try {
+    const config: AxiosRequestConfig = {
+      params: {
+        "api-version": "3",
+        "$filter": odsFilter
+      },
+      headers: {"apikey": `${effectiveApiKey}`}
+    }
+
+    const response = await axios.get(apigeeDoHSEndpoint, config)
+    logger.info("Successfully fetched DoHS API response", {data: response.data})
+
+    // Map DoHS response to correct roles
+    const mappedData: Record<string, {ODSCode: string} | null> = {
+      prescribingOrganization: response.data.value.find((item: {ODSCode: string}) =>
+        item.ODSCode === odsCodes.prescribingOrganization) ?? null,
+      nominatedPerformer: response.data.value.find((item: {ODSCode: string}) =>
+        item.ODSCode === odsCodes.nominatedPerformer) ?? null,
+      dispensingOrganizations: response.data.value.filter((item: {ODSCode: string}) =>
+        odsCodes.dispensingOrganizations?.includes(item.ODSCode)) ?? []
+    }
+
+    return mappedData
+  } catch (error) {
```

don't bother catching and logging errors, just let them bubble up